packages/deployed-backend-client/src/backend_output_client.ts

```diff
@@ -22,6 +27,64 @@ export class DefaultBackendOutputClient implements BackendOutputClient {
       this.amplifyClient
     ).getStrategy(backendIdentifier);
     const output = await outputFetcher.fetchBackendOutput();
-    return unifiedBackendOutputSchema.parse(output);
+
+    return unifiedBackendOutputSchema.parse({
+      ...output,
+      ...(output[storageOutputKey] && {
+        [storageOutputKey]: this.parseStorageOutput(output[storageOutputKey]),
```

Comment:
it is odd to have storage output parsing being so specific.
can't this be solved by modeling schema properly?

Reply:
+1 this should all be handled by zod if we model the type properly

Reply:
Coming back to this, I think you may be conflating reading backend output and formatting client config. This class should only be responsible for reading stack output and metadata and reconstituting it into a validated and typed object. Then the client config contributors take that object and format it however is needed. That happens here for storage: https://github.com/aws-amplify/amplify-backend/blob/7361ff4ed9da10f9b48e327c96eff25eb81a9e10/packages/client-config/src/client-config-contributor/client_config_contributor_v1.ts#L254